src/App/Multisite/MultisiteManager.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/**
+ * Ushahidi Platform
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Platform
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\App\Multisite;
+
+use Illuminate\Contracts\Events\Dispatcher;
+
+class MultisiteManager
```

Comment:
This is probably a good place to start reviewing. This class mostly replaces the old `Multisite` class but split into more sensible pieces